src/engine/strat_engine/filesystem.rs

```diff
@@ -56,6 +59,52 @@ impl StratFilesystem {
         }
     }
 
+    /// Create a snapshot of the ThinDev under the filesystem. Return the
+    /// resulting ThinDev to the caller.
+    pub fn snapshot(&self,
+                    dm: &DM,
+                    thin_pool: &ThinPoolDev,
+                    snapshot_name: &DmName,
+                    snapshot_fs_uuid: FilesystemUuid,
+                    snapshot_thin_id: ThinDevId)
+                    -> EngineResult<ThinDev> {
+
+        match self.thin_dev
+                  .snapshot(dm, thin_pool, snapshot_name, snapshot_thin_id) {
+            Ok(thin_dev) => {
+                // If the source is mounted, XFS puts a dummy record in the
+                // log to enforce replay of the snapshot to deal with any
+                // orphaned inodes. The dummy record put the log in a dirty
+                // state. xfs_admin won't allow a filesystem UUID
+                // to be updated when the log is dirty.  To clear the log
+                // we mount/unmount the filesystem before updating the UUID.
+                //
+                // If the source is unmounted the XFS log will be clean so
+                // we can skip the mount/unmount.
+                if let Some(_origin_mount_point) = self.get_mount_point()? {
```

Comment:
This is more minor, but since you're really not interested in the value

```
if self.get_mount_point()?.is_some() {
    ...
```

is better.

Reply:
Ok